### PR TITLE
feat(query): support interval function

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

- fix: #17060

## Tests

- [x] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/databendlabs/databend/17093)

### CODE REVIEW

File: src/query/functions/src/scalars/comparison.rs [hunk 1]
```diff
@@ -79,6 +81,39 @@ const ALL_FALSE_DOMAIN: BooleanDomain = BooleanDomain {
     has_false: true,
 };
 
+fn register_interval_cmp(registry: &mut FunctionRegistry) {
+    registry.register_comparison_2_arg::<IntervalType, IntervalType, _, _>(
```
Comment (sundy-li) [inline src/query/functions/src/scalars/comparison.rs:85]: maybe `register_simple_domain_type_cmp`

File: src/query/functions/src/scalars/interval.rs [hunk 2]
```diff
@@ -79,3 +86,227 @@ fn register_interval_to_string(registry: &mut FunctionRegistry) {
...
+
+    registry
+        .register_passthrough_nullable_2_arg::<TimestampType, IntervalType, TimestampType, _, _>(
+            "plus",
+            |_, _, _| FunctionDomain::MayThrow,
+            vectorize_with_builder_2_arg::<TimestampType, IntervalType, TimestampType>(
+                |a, b, output, ctx| {
+                    // plus microseconds and days
+                    let ts = a
+                        .wrapping_add(b.microseconds())
```
Comment (sundy-li) [inline src/query/functions/src/scalars/interval.rs:113]: why not use `total_micros `

Comment (TCeason) [reply src/query/functions/src/scalars/interval.rs:113]: Add total micros is not right. 

If interval compression 

30day is equal with 1 month.

But if timestamp add 30 days and add 1month maybe diff.

File: src/common/column/src/types/native.rs [hunk 3]
```diff
@@ -268,6 +267,33 @@ impl NativeType for days_ms {
 #[repr(C)]
 pub struct months_days_micros(pub i128);
 
+const MICROS_PER_DAY: i64 = 24 * 3600 * 1_000_000;
+const MICROS_PER_MONTH: i64 = 30 * MICROS_PER_DAY;
```
Comment (b41sh) [inline src/common/column/src/types/native.rs:271]: The number of days in each month is different, can we use a fixed number?

Comment (TCeason) [reply src/common/column/src/types/native.rs:271]: > The number of days in each month is different, can we use a fixed number?

Well, there's no need to think about it. The same is true in other implementations of the interval type of db. Like duckdb and snowflake

File: src/query/functions/src/scalars/comparison.rs (hunk 1 above)
Comment (TCeason) [reply src/query/functions/src/scalars/comparison.rs:85]: Yes.

**Review by b41sh — APPROVED** [review]

**Review by sundy-li — APPROVED** [review]

**Pull request merged**